app.py

```diff
@@ -19,17 +20,28 @@ def home() -> str:
 
 @app.route('/animals', methods=['GET'])
 def index() -> Response:
-    animals = Animal.query.all()
-    return jsonify({"animals": [AnimalResponse.model_validate(animal).model_dump(mode='json') for animal in animals]})
+    name = request.args.get('name')
+    animals = Animal.query.filter_by(name=name).all() if name else Animal.query.all()
```

Comment:
You can filter by substring (part of name) in this way
`Model.query.filter(Model.columnName.contains('sub_string'))`